applications/StructuralMechanicsApplication/custom_conditions/small_displacement_line_load_condition.cpp

```diff
@@ -0,0 +1,231 @@
+
+template<std::size_t TDim>
+Condition::Pointer SmallDisplacementLineLoadCondition<TDim>::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties
+    ) const
+{
+    return Kratos::make_intrusive<SmallDisplacementLineLoadCondition<TDim>>( NewId, this->GetGeometry().Create(ThisNodes), pProperties );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+Condition::Pointer SmallDisplacementLineLoadCondition<TDim>::Clone (
+    IndexType NewId,
+    NodesArrayType const& ThisNodes
+    ) const
+{
+    KRATOS_TRY
+
+    Condition::Pointer p_new_cond = this->Create(NewId, ThisNodes, this->pGetProperties());
+    p_new_cond->SetData(this->GetData());
+    p_new_cond->Set(Flags(*this));
+    return p_new_cond;
+
+    KRATOS_CATCH("");
+}
+
+
+//******************************* DESTRUCTOR *****************************************
+/***********************************************************************************/
+
+template<std::size_t TDim>
+SmallDisplacementLineLoadCondition<TDim>::~SmallDisplacementLineLoadCondition()
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+void SmallDisplacementLineLoadCondition<TDim>::CalculateAll(
+    MatrixType& rLeftHandSideMatrix,
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo,
+    const bool CalculateStiffnessMatrixFlag,
+    const bool CalculateResidualVectorFlag
+    )
+{
+    KRATOS_TRY;
+
+    const auto& r_geometry = this->GetGeometry();
+    const SizeType number_of_nodes = r_geometry.size();
+    const SizeType block_size = this->GetBlockSize();
+
+    // Resizing as needed the LHS
+    const SizeType mat_size = number_of_nodes * block_size;
+
+    // Resizing as needed the RHS
+    if ( CalculateResidualVectorFlag ) { // Calculation of the matrix is required
+        if ( rRightHandSideVector.size( ) != mat_size ) {
+            rRightHandSideVector.resize( mat_size, false );
+        }
+        noalias( rRightHandSideVector ) = ZeroVector( mat_size ); //resetting RHS
+    }
+
+
+    // Reading integration points and local gradients
+    const auto integration_method = IntegrationUtilities::GetIntegrationMethodForExactMassMatrixEvaluation(r_geometry);
+    const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints(integration_method);
+    const Matrix& rNcontainer = r_geometry.ShapeFunctionsValues(integration_method);
+
+    // Sizing work matrices
+    Vector pressure_on_nodes = ZeroVector( number_of_nodes );
+
+    // Pressure applied to the element itself
+    double pressure_on_condition = 0.0;
```

Comment:
i think you moved this to a static utility, am i wrong?

Reply:
The corresponding parts to the solid, not the loads. I already derived and the code duplicated is the minimum